deepchem/models/tests/test_infograph.py

```diff
@@ -0,0 +1,103 @@
+import os
+import pytest
+import numpy as np
+import deepchem as dc
+from deepchem.feat.molecule_featurizers import MolGraphConvFeaturizer
+
+
+@pytest.mark.torch
+def get_dataset(mode='classification'):
```

Comment:
Let's split this up into two simpler helper functions, one for classification and one for regression? A good rule of thumb is that a unit test shouldn't take thought to read through. This helper is complex enough it requires additional thought